src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonService.kt

```diff
@@ -2,23 +2,49 @@ package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.servic
 
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClientResponseException
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.client.VisitSchedulerClient
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.visit.scheduler.DateRange
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.visit.scheduler.PrisonDto
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.visit.scheduler.enums.UserType
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.exception.NotFoundException
 import java.time.LocalDate
 
 @Service
 class PrisonService(
   private val visitSchedulerClient: VisitSchedulerClient,
 ) {
   companion object {
-    val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+    val logger: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-  fun getPrison(prisonCode: String): PrisonDto {
-    return visitSchedulerClient.getPrison(prisonCode)!!
+  fun getPrison(prisonCode: String): PrisonDto? {
+    var prison: PrisonDto? = null
+
+    try {
+      prison = visitSchedulerClient.getPrison(prisonCode)
+    } catch (e: WebClientResponseException) {
+      logger.info("Failed to get details for prison - $prisonCode from visit-scheduler, error = ${e.message}")
+      if (e.statusCode != HttpStatus.NOT_FOUND) {
```

Comment:
Could we throw the NotFound exception here and remove the need to return a null?

Reply:
I would say this logic visitSchedulerClient should be concerned with HTTP errors and connection errors. In others please move into the client.

Reply:
implemented both suggestions.

Reply:
Yeha I understand why the PrisonDto has to be PrisonDto? that time out right.

Reply:
yes in case there is a timeout error